src/mongoose_ldap_worker.erl

```diff
@@ -0,0 +1,148 @@
+-module(mongoose_ldap_worker).
+
+-behaviour(gen_server).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+-include("mongoose.hrl").
+-include("eldap.hrl").
+
+-type state() :: #{handle := none | eldap:handle(),
+                   servers := [string()],
+                   encrypt := none | tls,
+                   tls_options := list(),
+                   port := pos_integer(),
+                   root_dn := binary(),
+                   password := binary(),
+                   connect_interval := pos_integer()}.
+-type request() :: {function(), Args :: [any()]}.
+
+%% gen_server callbacks
+
+-spec init(list()) -> {ok, state()}.
+init(Options) ->
+    State = initial_state(Options),
+    self() ! connect,
```

Comment:
Because we do not want to wait until it connects - the same happens if the connection breaks, we also reconnect asynchronously in that case. The rule is that the worker stays alive regardless of the connection state.